proto/contract.proto

```diff
@@ -297,6 +294,10 @@ message CloudEventOverrides {
   map<string, string> extensions = 1;
 }
 
+message FeatureFlags {
+  bool enableEventTypeAutocreate = 1;
+}
```

Comment:
@creydr wondering if we should pass the full list here as a Map ? and perhaps call the field in `Resource` `EventingCoreFeatureFlags` to distinguish between core and Kafka ones (in case we need them in the future)

Reply:
I was thinking that would be good in the future, I just wasn't 100% sure about the approach so I decided to keep it minimal when starting on this PR. Happy to add the full set here/in a follow up PR if it makes sense

Reply:
Having the full set could also help us on some OIDC stuff to get rid of the config-features CM mount point and watchers.

Reply:
only one question from my side: is there a reason, why we would place it in the `Resource` message type, instead of at the top level on the `Contract` (similar to the trust bundles), as they should not really change between resources

Reply:
This was because for EventType autocreate we sometimes want to disable it for a resource even when it is enabled. Specifically, if a channel is owned by a MTChannelBasedBroker, we should not create eventtypes on the channel as they will already be created by the broker

Reply:
I like that additional flexibility on the resource level